src/highlight/highlightUtil.ts

```diff
@@ -22,30 +27,41 @@ export function getHighlightArea(shapes: Rect[]): number {
 
 export function getRange(): Range | null {
     const selection = window.getSelection();
-    if (!selection || selection.isCollapsed) {
+    if (!selection || selection.type !== 'Range' || selection.isCollapsed) {
         return null;
     }
 
     return selection.getRangeAt(0);
 }
 
+export const DOMRectToRect = ({ height, width, x, y }: DOMRect): Rect => ({
+    height,
+    type: 'rect',
+    width,
+    x,
+    y,
+});
+
 export function getSelectionItem(): SelectionItem | null {
     const range = getRange();
 
     if (!range) {
         return null;
     }
 
-    const { page: startPage } = getPageInfo(range.startContainer as Element);
-    const { page: endPage } = getPageInfo(range.endContainer as Element);
+    const startPage = getPageNumber(range.startContainer as Element);
+    const endPage = getPageNumber(range.endContainer as Element);
 
-    if (startPage < 1 || endPage < 1 || startPage !== endPage) {
+    if (!startPage || !endPage || startPage !== endPage) {
         return null;
     }
 
-    const { height, width, x, y } = range.getBoundingClientRect();
+    const boundingRect = DOMRectToRect(range.getBoundingClientRect());
+    const rects = Array.from(range.getClientRects())
+        .filter(({ height, width, x, y }) => x > 0 && y > 0 && height < window.innerHeight && width < window.innerWidth)
```

Comment:
Under what circumstances are these conditions possible? Can we avoid referring to `window`?

Reply:
I remember if the whole page is selected, some malformed rects will be included in selection. But I saw in https://github.com/box/box-annotations/pull/553, the malformed rects would be filtered, so I will remove this line.